### PR TITLE
test(backend): fix TestJobApis test is flaky. Fixes #4419

### LINKED ISSUE #4419: Testing - Backend - The TestJobApi/TestJobApis test is not stable
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/4415/kubeflow-pipeline-e2e-test/1298513122180468740
```
integration-test-k95mb-63616182: === RUN   TestJobApi/TestJobApis
integration-test-k95mb-63616182: --- FAIL: TestJobApi (83.28s)
integration-test-k95mb-63616182:     --- FAIL: TestJobApi/TestJobApis (41.45s)
integration-test-k95mb-63616182:         job_api_test.go:177: 
integration-test-k95mb-63616182:             	Error Trace:	job_api_test.go:177
integration-test-k95mb-63616182:             	Error:      	Not equal: 
integration-test-k95mb-63616182:             	            	expected: "hello world"
integration-test-k95mb-63616182:             	            	actual  : "argument parameter"
integration-test-k95mb-63616182:             	            	
integration-test-k95mb-63616182:             	            	Diff:
integration-test-k95mb-63616182:             	            	--- Expected
integration-test-k95mb-63616182:             	            	+++ Actual
integration-test-k95mb-63616182:             	            	@@ -1 +1 @@
integration-test-k95mb-63616182:             	            	-hello world
integration-test-k95mb-63616182:             	            	+argument parameter
integration-test-k95mb-63616182:             	Test:       	TestJobApi/TestJobApis
integration-test-k95mb-63616182:         job_api_test.go:183: 
integration-test-k95mb-63616182:             	Error Trace:	job_api_test.go:183
integration-test-k95mb-63616182:             	Error:      	Not equal: 
integration-test-k95mb-63616182:             	            	expected: "argument parameter"
integration-test-k95mb-63616182:             	            	actual  : "hello world"
integration-test-k95mb-63616182:             	            	
integration-test-k95mb-63616182:             	            	Diff:
integration-test-k95mb-63616182:             	            	--- Expected
integration-test-k95mb-63616182:             	            	+++ Actual
integration-test-k95mb-63616182:             	            	@@ -1 +1 @@
integration-test-k95mb-63616182:             	            	-argument parameter
integration-test-k95mb-63616182:             	            	+hello world
integration-test-k95mb-63616182:             	Test:       	TestJobApi/TestJobApis
```

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
We list by created_at time, but its precision is 1 second.

Fixes #4419

### CODE REVIEW

File: backend/test/integration/upgrade_test.go [hunk 1]
```diff
@@ -108,15 +108,16 @@ func (s *UpgradeTests) SetupSuite() {
...
-		t := s.T()
-
-		// Clean up after the suite to unblock other tests. (Not needed for upgrade
-		// tests because it needs changes in prepare tests to persist and verified
-		// later.)
-		test.DeleteAllExperiments(s.experimentClient, t)
-		test.DeleteAllPipelines(s.pipelineClient, t)
-		test.DeleteAllRuns(s.runClient, t)
-		test.DeleteAllJobs(s.jobClient, t)
+		if !*isDevMode {
```
Comment (Bobgy) [inline backend/test/integration/upgrade_test.go:111]: this flag is only set when running locally, it preserves data after test suites finish

**Comment by numerology** [comment]
/lgtm

Thanks @Bobgy

**Pull request merged**